pkg/controller/clusterapi.go

```diff
@@ -91,3 +91,18 @@ func GetMachineDeployment(ctx context.Context, client client.Client, machineDepl
 	}
 	return machineDeployment, nil
 }
+
+// GetMachineDeployments reads all of cluster-api MachineDeployment for an eks-a cluster using a kube client.
+func GetMachineDeployments(ctx context.Context, c client.Client, cluster *anywherev1.Cluster) ([]clusterv1.MachineDeployment, error) {
+	machineDeployments := &clusterv1.MachineDeploymentList{}
+
+	err := c.List(ctx, machineDeployments, client.MatchingLabels{clusterv1.ClusterNameLabel: cluster.Name}, client.InNamespace(constants.EksaSystemNamespace))
+	if apierrors.IsNotFound(err) {
+		return nil, nil
```

Comment:
we should test this since it's part of the main behavior this method proposes

Reply:
This one should not be here actually :)